ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,32 +31,216 @@ public  void Init()
         }
         public  void Run()
         {
+            
             WriteLine("Welcome to Reverse Tic Tac Toe!" + Environment.NewLine);
-            getBoardSizeFromUser();
-            getGameTypeFromUser();
-            m_Game = new Game(m_BoardSize, m_GameType);
-            while(true)
+            int BoardSize =  getBoardSizeFromUser();
+            Game.eGameTypes GameType = getGameTypeFromUser();
+            m_Game = new Game(BoardSize, GameType);
+           // while(true)
+            //{
+            printGameBoard();
+            playTurn();
+            ReadLine();
+            
+            //}
+        }
+
+        private void playTurn()
+        {
+            int row = 0;
+            int col = 0;
+            //int return status: continue, place_taken, finished game
+            
+            if(m_Game.CurrentTurn == 1 || m_Game.GameType == Game.eGameTypes.PVP)
+            {
+Label: 
+                getRowColFromUser(out row, out col);
+                string msg ="";
+                int retStatus;
+                if (row == QUIT) // if press Q
+                {
+                    m_Game.Playerforfeit();
+                     msg = getEndGameMsg(Game.StatusWin);
+                }
+                // if Place is taken/ win/ tie
+                else if(!m_Game.PlaceSymbolAndReturnStatus(row, col, out retStatus))
+                {
+                    if(retStatus == Game.StatusTaken)
+                    {
+                        WriteLine("Place is taken!");
+                        goto Label;
```

Comment:
Do you have to use "goto" here? Guy said in class that using that is considered a bad practice, and reduces readabillity, and so we use it only when there's absolutly no choice.

Reply:
Instead of the "labe"l and "goto",  maybe you can use a while loop:
while(!moveMadeSuccessfully)
{
    if (!AttemptMove)
    {
        WriteLine("move failed due to" + reason + ", Please try again:  ....);
        getMoveInput();
    }
    else
    {
        moveMadeSuccessfully = true;
    }
}